appservice/src/SiteWrapper.ts

```diff
@@ -293,6 +298,43 @@ export class SiteWrapper {
         await requestP.get(`https://${this.defaultHostName}/admin/host/status`, requestOptions);
     }
 
+    public async validateDeployment(outputChannel: vscode.OutputChannel, pollingInterval: number = 5000, pollingAttempts: number = 12): Promise<responseStatus[]> {
+        // by default, try for a minute before we declare that the deployment failed
+        // tslint:disable-next-line:no-http-string
+        const uri: string = `${this.isSsl ? 'https://' : 'http://'}${this.defaultHostName}`;
+        const validatingDeployment: string = `Validating your ${this.isFunctionApp ? 'function' : 'web'} app successfully deployed.`;
+        const options: {} = {
+            method: 'GET',
+            uri: uri,
+            resolveWithFullResponse: true
+        };
+        let response: IncomingMessage;
+        const statuses: responseStatus[] = [];
+        for (let i: number = 0; i < pollingAttempts; i += 1) {
+            try {
+                await new Promise((resolve: () => void): void => { setTimeout(resolve, pollingInterval); });
+                // wait for the polling interval to send a request
+                this.log(outputChannel, validatingDeployment);
+                // tslint:disable-next-line:no-unsafe-any
+                response = <IncomingMessage>(await requestP(options));
+                // request throws an error for 400-500 responses
+                if (response.statusCode >= 200 && response.statusCode < 400) {
+                    // a status code 200-300 indicates success
+                    statuses.push({ code: response.statusCode, message: response.statusMessage });
+                    return statuses;
```

Comment:
Based on the name of this function, I would expect it to return void and throw an error if it fails to validate. If you don't want to throw an error, then rename it to "tryValidateDeployment" ('try' usually indicates a function will not throw an error).

I'm also not sure what you expect the consumer of this method to do with the array of status codes? IMO only the last status code is useful